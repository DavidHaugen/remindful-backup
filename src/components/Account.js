import React, { Component } from 'react'
import remindfulApiService from '../services/remindful-api-service';
import TokenService from '../services/token-service'
import GoalsContext from '../context/GoalsContext'

class Account extends Component {
  static contextType = GoalsContext

  state = {
    deleted: false
  }

  deleteAccount = () => {
    this.context.clearError();
    remindfulApiService.deleteUser()
    .catch(res => {
      this.context.setError(res)
    })
    .then(TokenService.clearAuthToken())
      .then(this.context.deleteUser())
      .then(this.props.history.push('/account-deleted'))
  }
  render(){

    if(this.context.testUser){
      return(
        <div className="main" role="main">
          <div className="wrapper">
           <p className="description">We really like our test account, so we have limited what you can see on this page. If you really want to delete an account, go make one and come on back.</p>
          </div>
        </div>
      )
    } else {
      return (
        <div className="main" role="main">
          <button className="textButton" onClick={() => this.deleteAccount()}>Delete my account</button>
        </div>
      )
    }
  }
}

export default Account;