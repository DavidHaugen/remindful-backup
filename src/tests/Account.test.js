import React from 'react';
import {MemoryRouter} from 'react-router-dom'
import Account from '../components/Account'
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

describe('<Account />', () => {
  it('Renders without crashing', () => {
      mount(<MemoryRouter><Account/></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><Account/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});

