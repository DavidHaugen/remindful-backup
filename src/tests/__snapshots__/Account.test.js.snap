// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> renders the UI as expected 1`] = `
<div
  className="main"
  role="main"
>
  <button
    className="textButton"
    onClick={[Function]}
  >
    Delete my account
  </button>
</div>
`;
